src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -432,6 +433,25 @@ Copyright (C) 2011-2012 Xamarin. All rights reserved.
   <FilterAssemblies InputAssemblies="@(_ReferenceDependencyPaths)">
     <Output TaskParameter="OutputAssemblies" ItemName="_MonoAndroidReferenceDependencyPaths" />
   </FilterAssemblies>
+  <MakeDir
+      Condition=" '@(_AndroidMamMappingFile->Count())' != '0' "
+      Directories="$(MonoAndroidAssetsDirIntermediate)/xa-internal"
+  />
+  <PropertyGroup>
+    <_AndroidMamMappingXml>$(MonoAndroidAssetsDirIntermediate)xa-internal/xa-mam-mapping.xml</_AndroidMamMappingXml>
+  </PropertyGroup>
+  <MamJsonToXml
+      Condition=" '@(_AndroidMamMappingFile->Count())' != '0' "
+      MappingFiles="@(_AndroidMamMappingFile)"
+      XmlMappingOutput="$(_AndroidMamMappingXml)"
+  />
+  <ItemGroup
+      Condition=" Exists ('$(_AndroidMamMappingXml)') ">
+    <AndroidAsset Include="$(_AndroidMamMappingXml)" LogicalName="xa-internal/xa-mam-mapping.xml" />
+  </ItemGroup>
+  <ItemGroup>
+    <FileWrites Include="$(_AndroidMamMappingXml)" />
+  </ItemGroup>
```

Comment:
Could this chunk of MSBuild be in its own target? I'm thinking we'd want a `Condition` that skips the target:

```xml
Condition=" '@(_AndroidMamMappingFile->Count())' != '0' "
```

And then maybe the Inputs/Outputs would be:
```xml
Inputs="$(_AndroidBuildPropertiesCache);@(_AndroidMamMappingFile)"
Outputs="$(_AndroidMamMappingXml)"
```

Reply:
I wish i could figure out how to have a "nice, short, readable" `Outputs` value, instead of the "full path" I'm currently using, but otherwise… this has been done.  More targets, smaller targets.